api/envoy/extensions/filters/http/oauth/v3/oauth.proto

```diff
@@ -0,0 +1,68 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.oauth.v3;
+
+import "google/protobuf/duration.proto";
+
+import "udpa/annotations/status.proto";
+import "udpa/annotations/versioning.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.oauth.v3";
+option java_outer_classname = "OauthProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: OAuth]
+// OAuth :ref:`configuration overview <config_http_filters_oauth>`.
+// [#extension: envoy.filters.http.oauth]
+//
+
+message OAuth2Credentials {
+  // The client_id and client_secret should be provided by the OAuth provider.
+  string client_id = 1;
+
+  string client_secret = 2;
+
+  string token_secret = 3;
```

Comment:
shouldn't these marked with sensitive annotation?

Reply:
No, they are just the keys for those secrets. It'll be clear once I add comments :-)